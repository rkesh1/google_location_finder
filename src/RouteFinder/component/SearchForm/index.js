import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Autocomplete from '../AutoComplete';
import Button from '../Button';
import './index.scss';

class SearchForm extends Component{
  constructor(props){
    super(props);
    this.state = {
      formData: {
        destination: '',
        source: ''
      },
      disabled: true,
    }
  }

   /**
   * @method showPlaceDetails
   * @description Get data from input field and stored in state 
   * @param {string} name is used to input field name
   * @param {string} value is return from Autocomplete value
   */
  showPlaceDetails = (name, value) => {
    const {formData} = this.state;
    const newObj = {...formData, [name]:value};
    const {source, destination} = newObj;
    const disabled = source === '' || destination === '' || value === '';
    this.setState({
      formData: {...formData, [name]:value},
      disabled
    });
  }

  /**
   * @method submitForm
   * @description Submit form handler submit form data  
   */
  submitForm = () => {
    const {submitFormHandler} = this.props;
    const {formData} = this.state;
    submitFormHandler(formData);
  }

  

  render(){
    const {formData, disabled} = this.state;
    const {source, destination} = formData;
    const {message, clearTextField, resetFormCompleted, resetFormHandler, resubmit, clearMessage }  = this.props;
    const commonProps = {
      onPlaceChanged: this.showPlaceDetails,
      clearTextField,
      resetFormCompleted,
      clearMessage
    }
    const inputField = [{
      labelTitle: 'Starting location',
      name: 'source',
      value: source,
      id: 'source',
      type: 'text',
      placeholder: 'Starting location',
      ...commonProps
    },
    {
      labelTitle: 'Drop-off point',
      name: 'destination',
      value: destination,
      id: 'destination',
      type: 'text',
      placeholder: 'Drop-off point',
      ...commonProps
    }
  ];

    const submitButton = {
      value: resubmit ? 'Re-Submit': 'Submit',
      clickHandler: this.submitForm,
    }
    
    const {showErr, errorResponse, renderPath, totalDistance, totalTime} = message;
    return (
      <div className="formContainer">
        { inputField.map((item,key) => (
          <Autocomplete {...item}  key={key}/>
        ))}
      {showErr && 
        <div className="error-container">
          <p>{errorResponse}</p>
        </div>
        }
        {renderPath && 
        <div className="info">
          <p>total distance: {totalDistance}</p>
          <p>total time: {totalTime}</p>
        </div>
        }
      <div className="form-element form-controller">
        <Button {...submitButton} disabled={disabled} additionClass="primary-btn"/>
        <Button {...submitButton} value="Reset" clickHandler={resetFormHandler} />
      </div>
    </div>
    );
  }
  
}

SearchForm.propTypes = {
  submitFormHandler: PropTypes.func.isRequired,
  resetFormHandler: PropTypes.func.isRequired,
  clearTextField: PropTypes.bool,
  message: PropTypes.object,
  resubmit:PropTypes.bool,
  clearMessage:PropTypes.func,
};

SearchForm.defaultProps = {
  message: {},
  clearTextField: false,
  resubmit: false,
  clearMessage: ()=>{},
};
export default SearchForm;
