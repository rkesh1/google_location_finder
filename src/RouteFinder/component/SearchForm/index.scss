  .formContainer {
    padding: 20px;

    & .form-element {
        margin-bottom: 20px;

        &.form-controller{
          margin-top: 40px;
        }
      }

      & .error-container{
        padding: 20 0px;
        & p{
          color: red;
          font-weight: 700;
        }
      }
      & .info{
        padding: 20 0px;
      }
      
  }

  

  

  