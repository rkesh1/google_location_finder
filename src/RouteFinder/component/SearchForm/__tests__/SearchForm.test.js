import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from '../index';


describe('<SearchForm /> Component Test Case ', () => {
  let wrapper;
  const props = {
    submitFormHandler: () => { },
    resetFormHandler: () => { },
    clearTextField: false,
    message: 'input',
  }
  /**
   * Renders a shallow version of the SearchForm
   */
  function renderShallow() {
    wrapper = shallow(<SearchForm {...props} />);
  }


  it('SearchForm Component render', () => {
    renderShallow();
    expect(wrapper).toMatchSnapshot();
  });

  
  it("change in form field", async () => {
    renderShallow();
    const instance = wrapper.instance();
    await instance.showPlaceDetails('xyz', '123');
    expect(instance.state.formData.xyz).toBe('123');
  });

  it("DOM varified on based of props", async () => {
    const message={
      renderPath: true,
      showErr: true,
    }
    wrapper = shallow(<SearchForm {...props} message={message} />);
    expect(wrapper.find('.info').length).toBe(1);
    expect(wrapper.find('.error-container').length).toBe(1);
  });
});
