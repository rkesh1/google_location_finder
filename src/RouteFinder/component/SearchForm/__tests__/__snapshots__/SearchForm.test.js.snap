// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> Component Test Case  SearchForm Component render 1`] = `ShallowWrapper {}`;
